using_images/db_images/mysql.adoc

```diff
@@ -694,3 +694,62 @@ When scaling down, the MySQL slave is shut down and, because the slave does not
 have any persistent storage defined, all data on the slave is lost. The MySQL
 master server then discovers that the slave is not reachable anymore, and it
 automatically removes it from the replication.
+
+[[troubleshooting]]
+== Troubleshooting
+
+// TODO: Put tags around the body of this section so that
+// it can be snarfed by the OSE Troubleshooting Guide.
+
+This section describes some troubles you might encounter
+and presents possible resolutions.
+
+[[linux-native-aio-failure]]
+=== Linux Native AIO Failure
+
+.Symptom
+The MySQL container fails to start and the logs show something like:
+
+----
+151113  5:06:56 InnoDB: Using Linux native AIO
+151113  5:06:56  InnoDB: Warning: io_setup() failed with EAGAIN. Will make 5 attempts before giving up.
+InnoDB: Warning: io_setup() attempt 1 failed.
+InnoDB: Warning: io_setup() attempt 2 failed.
+Waiting for MySQL to start ...
+InnoDB: Warning: io_setup() attempt 3 failed.
+InnoDB: Warning: io_setup() attempt 4 failed.
+Waiting for MySQL to start ...
+InnoDB: Warning: io_setup() attempt 5 failed.
+151113  5:06:59  InnoDB: Error: io_setup() failed with EAGAIN after 5 attempts.
+InnoDB: You can disable Linux Native AIO by setting innodb_use_native_aio = 0 in my.cnf
+151113  5:06:59 InnoDB: Fatal error: cannot initialize AIO sub-system
+151113  5:06:59 [ERROR] Plugin 'InnoDB' init function returned error.
+151113  5:06:59 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
+151113  5:06:59 [ERROR] Unknown/unsupported storage engine: InnoDB
+151113  5:06:59 [ERROR] Aborting
+----
+
+.Explanation
+MySQL's storage engine was unable to use of the kernel's
+AIO (Asynchronous I/O) facilities due to resource limits.
+
+.Resolution
+
+1. Turn off AIO usage entirely,
+by setting environment variable `MYSQL_AIO` to have value `0`.
+On subsequent deployments, this arranges for the
+MySQL configuration variable `innodb_use_native_aio`
+to have value `0`.
+
+1. Increase the `aio-max-nr` kernel resource.
+The following example examines the current value of `aio-max-nr` and doubles it.
++
+----
+$ sysctl fs.aio-max-nr
+fs.aio-max-nr = 1048576
+# sysctl -w fs.aio-max-nr=2097152
+----
++
+This is a per-node resolution and lasts until the next node reboot.
```

Comment:
Nice idea.  I tend to agree but am afraid we're moving out of scope.  I think it would be better to dig deeper in the Troubleshooting Guide (when it gets here -- see https://bugzilla.redhat.com/show_bug.cgi?id=1259118).